scripts/generate-crds/main.go

```diff
@@ -122,13 +122,22 @@ func generateTFBasedCRDs() []*apiextensions.CustomResourceDefinition {
 				log.Fatal(err)
 			}
 			crds := make([]*apiextensions.CustomResourceDefinition, 0)
+			directCount := 0
 			for _, rc := range rcs {
+				if rc.Direct {
+					fmt.Printf("skip generate TF-based CRD for direct resource %s\n", rc.Kind)
+					directCount += 1
+					continue
+				}
 				crd, err := crdgeneration.GenerateTF2CRD(&sm, rc)
 				if err != nil {
 					log.Fatalf("error generating CRD for %v: %v", rc.Name, err)
 				}
 				crds = append(crds, crd)
 			}
+			if directCount == len(rcs) {
```

Comment:
Actually I think we should do a check here to ensure RCs of different locationalities are either all direct or all non-direct.

So if directCount != len(rcs) then an error should be returned.

Reply:
Could you share more thoughts about "the RCs of different locationalities that have hybrid controllers"? How that could happen and what problems it could  cause to TF?

Reply:
Supporting the hybrid controllers means we need to solve conflicts for CRD generation and coordinate between controllers. So there will be some cost. There is no specific issue off the top of my head, but I expect there will be some.
In my opinion, a merged CRD should probably be supported by a single direct controller so context of one CRD can be found at a single place and it could help reduce confusion for maintainers. Though I guess certain level of confusion will always exist as it is not a one-on-one mapping, so this is not a strong argument.

If it is beneficial to support hybrid controllers (would like to know the reasons), then I'm not opinionated.